ksql-engine/src/main/java/io/confluent/ksql/util/QueryMetadata.java

```diff
@@ -70,6 +74,7 @@ public String getQueryApplicationId() {
   public void close() {
     kafkaStreams.close(100L, TimeUnit.MILLISECONDS);
     kafkaStreams.cleanUp();
+    kafkaTopicClient.deleteInternalTopics(queryApplicationId);
```

Comment:
Is this going to be called from each node? What is the behaviour if the topic is deleted by another node?
Also, for the line 75 above. I'm not sure waiting 100 milliseconds for streams to shutdown is long enough. It may be in the middle of processing etc, and writing to the various state stores and topics. So the cleanup and delete may well spew exceptions. It would be better to check the result of close. If it is false then you can use a loop to check the streams state, i.e, `KafkaStreams.state()` when it is `NOT_RUNNING` the instance has shutdown. Then you can call cleanup

Reply:
should also be exposed as a property

Reply:
I updated the behavior. After closing the stream job we will periodically check the status to see if the query is in NOT_RUNNING state for a configurable amount of time (default is 5 seconds). If we could not reach to NOT_RUNNING state by the end of this time we log an error message and move on.
Otherwise, we go ahead with the cleanup and deleting the internal topics.